test/api/model/initial_snapshot_test.dart

```diff
@@ -62,4 +63,15 @@ void main() {
       'unread_message_ids': [11, 2, 3],
     })).throws<AssertionError>();
   });
+
+  test('UserSettings.emojiset handles various unknown values', () {
+    final unknownValues = ['apple', 'microsoft', 'facebook', ''];
+    for (final unknownValue in unknownValues) {
+      final json = eg.userSettings().toJson()..['emojiset'] = unknownValue;
+      final settings = UserSettings.fromJson(json);
+
+      // Verify unknown emojiset defaults to Emojiset.unknown
+      check(settings.emojiset).equals(Emojiset.unknown);
```

Comment:
This comment doesn't add anything that isn't said equally clearly by the code. So best to leave it out:
```suggestion
      check(settings.emojiset).equals(Emojiset.unknown);
```